_docs/tasks/security/authn-policy.md

```diff
@@ -0,0 +1,288 @@
+
+```
+for from in "foo" "bar" "legacy"; do for to in "foo" "bar"; do kubectl exec $(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name}) -c sleep -n ${from} -- curl http://httpbin.${to}:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.${to}: %{http_code}\n"; done; done
+```
+
+```
+sleep.foo to httpbin.foo: 200
+sleep.foo to httpbin.bar: 200
+sleep.bar to httpbin.foo: 200
+sleep.bar to httpbin.bar: 200
+sleep.legacy to httpbin.foo: 000
+command terminated with exit code 56
+sleep.legacy to httpbin.bar: 200
+```
+
+## Enable mTLS for single service httpbin.bar.
+
+Run this command to set another policy for only for "httpbin" service on namespace "bar". Note in this example, we don't specify namespace in metadata but as part of the commandline (-n bar). They should work the same.
+
+```
+cat <<EOF | istioctl create -n bar -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "enable-mtls"
+spec:
+  targets:
+  - name: httpbin
+  peers:
+  - mtls:
+EOF  
+```
+
+Again, run the probing command. As expected, request from sleep.legacy to httpbin.bar starts failing.
+
+```
+...
+sleep.legacy to httpbin.bar: 000
+command terminated with exit code 56
+```
+
+If we have other services namespace bar, we should see traffic to them won't be affected. Instead of adding more services to demonstrate this behavior, we edit the policy sligthtly:
+
+```
+cat <<EOF | istioctl replace -n bar -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "enable-mtls"
+spec:
+  targets:
+  - name: httpbin
+    ports:
+    - number: 1234
+  peers:
+  - mtls:
+EOF
+```
+
+This new policy will apply only to httpbin on port 1234. As the result, mTLS is disabled (again) on port 8000 and request from sleep.legacy will success
+
+```
+kubectl exec $(kubectl get pod -l app=sleep -n legacy -o jsonpath={.items..metadata.name}) -c sleep -n legacy -- curl http://httpbin.bar:8000/ip -s -o /dev/null -w "%{http_code}\n"
+```
+
+```
+200
+```
+
+## Having both namespace-level and service-level policy.
+
+Assuming we already add the namespace-level policy that enable mTLS for all services in namespace foo and observe that request from sleep.legacy to httpbin.foo are failing (see above). Now add another policy for httpbin service directly. This poicy doesn't define any method for peer authentication, which equivalent to disable mTLS.
+
+```
+cat <<EOF | istioctl create -n foo -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "httpbin"
+spec:
+  targets:
+  - name: httpbin
+EOF
+```
+
+Re-run request from sleep.legacy, we should see success return code again (200), confirming service-level policy is overrule the namespace-level policy.
```

Comment:
success return code -> a success return code
is overrule -> overrules

Reply:
Done